### PR TITLE
Leverage generic persistence event classes

### PR DESCRIPTION
**Pull request by derrabus**
Follows doctrine/persistence#273

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
         "doctrine/inflector": "^1.4 || ^2.0",
         "doctrine/instantiator": "^1.3",
         "doctrine/lexer": "^1.2.3",
-        "doctrine/persistence": "^2.4",
+        "doctrine/persistence": "^2.5@dev",
```
Comment (derrabus) [inline composer.json]: TODO: Revert this live as soon as Persistence 2.5.0 has been released.

Comment (greg0ire) [reply composer.json]: Let's release persistence 2.5? Instead of risking forgetting about this / blocking 2.12.0?

Comment (derrabus) [reply composer.json]: I've released 2.5.

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Either we release 2.12.0 now or this should be retargeted to 2.11.x, as it fixes the build

**Comment by derrabus** [comment]
The build on 2.11.x has been fixed by #9648 already.

**Comment by michnovka** [comment]
Still getting Psalm errors

```
vendor/bin/psalm --show-info=false --stats --output-format=github
Target PHP version: 8.1 (set by config file)
Scanning files...
Analyzing files...

░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░  60 / 118 (50%)
░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░░
::error file=lib/Doctrine/ORM/Event/LifecycleEventArgs.php,line=31,col=16,title=MoreSpecificReturnType::lib/Doctrine/ORM/Event/LifecycleEventArgs.php:31:16: MoreSpecificReturnType: The declared return type 'Doctrine\ORM\EntityManagerInterface' for Doctrine\ORM\Event\LifecycleEventArgs::getEntityManager is more specific than the inferred return type 'Doctrine\Persistence\ObjectManager' (see https://psalm.dev/070)
::error file=lib/Doctrine/ORM/Event/LifecycleEventArgs.php,line=35,col=16,title=LessSpecificReturnStatement::lib/Doctrine/ORM/Event/LifecycleEventArgs.php:35:16: LessSpecificReturnStatement: The type 'Doctrine\Persistence\ObjectManager' is more general than the declared return type 'Doctrine\ORM\EntityManagerInterface' for Doctrine\ORM\Event\LifecycleEventArgs::getEntityManager (see https://psalm.dev/129)
::error file=lib/Doctrine/ORM/Event/LoadClassMetadataEventArgs.php,line=21,col=16,title=MoreSpecificReturnType::lib/Doctrine/ORM/Event/LoadClassMetadataEventArgs.php:21:16: MoreSpecificReturnType: The declared return type 'Doctrine\ORM\EntityManagerInterface' for Doctrine\ORM\Event\LoadClassMetadataEventArgs::getEntityManager is more specific than the inferred return type 'Doctrine\Persistence\ObjectManager' (see https://psalm.dev/070)
::error file=lib/Doctrine/ORM/Event/LoadClassMetadataEventArgs.php,line=25,col=16,title=LessSpecificReturnStatement::lib/Doctrine/ORM/Event/LoadClassMetadataEventArgs.php:25:16: LessSpecificReturnStatement: The type 'Doctrine\Persistence\ObjectManager' is more general than the declared return type 'Doctrine\ORM\EntityManagerInterface' for Doctrine\ORM\Event\LoadClassMetadataEventArgs::getEntityManager (see https://psalm.dev/129)
```

**Comment by derrabus** [comment]
You did upgrade to Persistence 2.5?

**Comment by michnovka** [comment]
```
composer show doctrine/persistence
name     : doctrine/persistence
descrip. : The Doctrine Persistence project is a set of shared interfaces and functionality that the different Doctrine object mappers share.
keywords : mapper, object, odm, orm, persistence
versions : * 2.5.0
type     : library
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
homepage : https://doctrine-project.org/projects/persistence.html
source   : [git] https://github.com/doctrine/persistence.git f8776dd9a0bdcd838812951a75f4ada72065a82a
dist     : [zip] https://api.github.com/repos/doctrine/persistence/zipball/f8776dd9a0bdcd838812951a75f4ada72065a82a f8776dd9a0bdcd838812951a75f4ada72065a82a
path     : /home/superuser/scripts/orm/vendor/doctrine/persistence
names    : doctrine/persistence

support
issues : https://github.com/doctrine/persistence/issues
source : https://github.com/doctrine/persistence/tree/2.5.0

autoload
psr-4
Doctrine\Common\ => src/Common
Doctrine\Persistence\ => src/Persistence

requires
doctrine/cache ^1.11 || ^2.0
doctrine/collections ^1.0
doctrine/deprecations ^0.5.3
doctrine/event-manager ^1.0
php ^7.1 || ^8.0
psr/cache ^1.0 || ^2.0 || ^3.0

requires (dev)
composer/package-versions-deprecated ^1.11
doctrine/annotations ^1.0
doctrine/coding-standard ^9.0
doctrine/common ^3.0
phpstan/phpstan ~1.4.10 || 1.5.0
phpunit/phpunit ^7.5.20 || ^8.5 || ^9.5
symfony/cache ^4.4 || ^5.4 || ^6.0
vimeo/psalm 4.22.0

conflicts
doctrine/annotations <1.0 || >=2.0
doctrine/common <2.10
```

**Comment by derrabus** [comment]
Okay, but the CI job is green. Something must be off on your side. https://github.com/doctrine/orm/runs/5970318458?check_suite_focus=true

**Comment by michnovka** [comment]
Yea, seems some cache issue.

```
vendor/bin/psalm --clear-cache
```

Fixed it. Sorry for wasting your time

**Pull request merged**